2017 Robot/src/org/usfirst/frc/team4999/lights/animations/AnimationSequence.java

```diff
@@ -28,13 +28,13 @@ public AnimationSequence(Animation[] animations, int[] times) {
 	}
 
 	@Override
-	public Color[] animate(Color[] pixels) {
+	public Packet[] animate() {
 		if(timepassed >= times[currentidx]) {
 			timepassed = 0;
 			currentidx++;
```

Comment:
Better written as:
currentidx = (currentidx + 1) % times.length;

I'm also struggling to understand the logic behind timepassed.  I think I see where you're heading, but there's quite a lot of awkwardness around how you're getting there.  This could also be a root cause of some of the glitches we were seeing on the looping.

Reply:
I was having the class add up the individual delays to find out how long the animation had been running for, then switching animation if the time running was >= the requested run time of the animation. It makes more sense, though, to store the time when the animation was started, and subtract it from the current time to see how long an animation has been running for.